### ISSUE TITLE
OWIN Saml2 'unexpected InResponseTo' in Sitecore

### ISSUE THREAD

**Issue by sitecorepm** [Sustainsys/Saml2#1125]
### Information needed
1. What nuget packages are you using
```
  <package id="Sustainsys.Saml2" version="1.0.0" targetFramework="net471" />
  <package id="Sustainsys.Saml2.Owin" version="1.0.0" targetFramework="net471" />
```

2. What is the expected behaviour
Able to login with IdP whether there is a ASP.NET session cookie or not

3. What happens instead.
Successful login only occurs when there is a ASP.NET session cookie

### Additional info
The Sitecore application I am attempting to integrate Sustainsys.Saml with is compiled against .NET 4.7.1 and that is also the latest version installed.

The details of my issue are detailed in this stackoverflow post:
https://sitecore.stackexchange.com/questions/22775/federated-authentication-with-saml2-issue

At this point, I've downloaded the Sustainsys source and debugged to the point that I know the Set-Cookie header being set from Sustainsys.Saml2.Owin.CommandResultExtensions.Apply but it doesn't get to the browser for some reason (according to browser DevTools and Fiddler). 

I'm not sure if this is a Sitecore issue or a Sustainsys issue but I was hoping for some advice on next steps to debug this. I'm running out of ideas.

Thanks!

**Comment by sitecorepm** [comment]
**Detailed troubleshooting/solution:** https://sitecorepm.wordpress.com/2019/11/01/sitecore-9-1-to-shibboleth-via-saml2/

**TLDR; answer**

This was due to a Katana (OWIN/Asp.Net) [known issue][1] that causes cookies to get dropped between OWIN and ASP.NET.

I solved it by adding the [Kentor.OwinCookieSaver][2] nuget package into my solution.


  [1]: https://github.com/aspnet/AspNetKatana/wiki/System.Web-response-cookie-integration-issues
  [2]: https://github.com/Sustainsys/owin-cookie-saver
